monkey/infection_monkey/i_puppet.py

```diff
@@ -0,0 +1,88 @@
+import abc
+import threading
+from collections import namedtuple
+from enum import Enum
+from typing import Dict, Optional, Tuple
+
+
+class PortStatus(Enum):
+    OPEN = 1
+    CLOSED = 2
+
+
+PortScanData = namedtuple("PortScanData", ["port", "status", "banner", "service"])
+
+
+class IPuppet(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def run_sys_info_collector(self, name: str) -> Dict:
+        """
+        Runs a system info collector
+        :param str name: The name of the system info collector to run
+        :return: A dictionary containing the information collected from the system
+        :rtype: Dict
+        """
+
+    @abc.abstractmethod
+    def run_pba(self, name: str, options: Dict) -> None:
+        """
+        Runs a post-breach action (PBA)
+        :param str name: The name of the post-breach action to run
+        :param Dict options: A dictionary containing options that modify the behavior of the PBA
+        """
+
+    @abc.abstractmethod
+    def ping(self, host: str) -> Tuple[bool, Optional[str]]:
+        """
+        Sends a ping (ICMP packet) to a remote host
+        :param str host: The domain name or IP address of a host
```

Comment:
`host` right now, in a lot of places we're using it, is an object of type `VictimHost`. Just pointing it out so we're sure this is going to be a string.

Reply:
Yeah. It's a string.